### PR TITLE
Fix/test case optimizations

### PR DESCRIPTION
**Pull request by himanshuchawla009**

### CODE REVIEW

File: packages/core/src/core.ts [hunk 1]
```diff
@@ -1062,6 +1126,9 @@ class ThresholdKey implements ITKey {
   }
 
   async inputShare(share: unknown, type?: string): Promise<void> {
+    if (!this.metadata) {
```
Comment (rathishubham7) [inline packages/core/src/core.ts:1129]: Why did you add these !this.metadata checks?? 
Is there a better way to do this ?

Comment (himanshuchawla009) [reply packages/core/src/core.ts:1129]: I added this coz if user call these functions before initializing then user should get appropriate error message. Currently these functions breaks with  "Cannot read property x of undefined message" when metadata is undefined. This check was already present in some functions.

**Review by YZhenY — APPROVED** [review]
lgtm
